### PR TITLE
Add ineffective sidecar message in ambient namespaces

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Resolve https://github.com/istio/istio/issues/48105

### CODE REVIEW

File: pkg/config/analysis/analyzers/util/in_mesh.go [hunk 1]
```diff
@@ -42,9 +42,20 @@ func PodInMesh(r *resource.Instance, c analysis.Context) bool {
...
+	}
 	return r.Metadata.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 }
 
+// NamespaceInAmbientMode returns true if a Namespace is configured as a ambient namespace.
+func NamespaceInAmbientMode(r *resource.Instance) bool {
+	if r == nil {
+		return false
+	}
+	return r.Metadata.Labels[constants.DataplaneMode] == constants.DataplaneModeAmbient
```
Comment (linsun) [inline pkg/config/analysis/analyzers/util/in_mesh.go:56]: Do we need to check if the namespace is also not have any sidecar injection label, e.g. istio.io/rev or istio-injection as either of these two labels win over our ambient label?

**Review by linsun — COMMENTED** [review]
LGTM, thank you @hanxiaop!!!

Left a small comment

File: pkg/config/analysis/analyzers/util/in_mesh.go (hunk 1 above)
Comment (hanxiaop) [reply pkg/config/analysis/analyzers/util/in_mesh.go:56]: Also, should we report a message regarding the mixed usage of sidecars and ambient pod in the same namespace?

File: pkg/config/analysis/analyzers/sidecar/selector.go [hunk 2]
```diff
@@ -116,4 +138,27 @@ func (a *SelectorAnalyzer) Analyze(c analysis.Context) {
 			c.Report(gvk.Sidecar, m)
 		}
 	}
+
+	for ns, sList := range namespacesToSidecars {
+		nsResource := namespaces[ns.String()]
+		// We don't want to report errors for namespaces in ambient mode, since there is no sidecar,
+		// but we do want to warn that the policy is ineffective.
+		// TODO: do we need to check mixed mode?
```
Comment (GregHanson) [inline pkg/config/analysis/analyzers/sidecar/selector.go:146]: agree we will probably have to revisit this scenario in the future

**Review by GregHanson — APPROVED** [review]
LGTM

File: pkg/config/analysis/analyzers/util/in_mesh.go (hunk 1 above)
Comment (linsun) [reply pkg/config/analysis/analyzers/util/in_mesh.go:56]: My preference is to not report Sidecar resource warning if the namespace also has sidecar injection labels, as those Sidecar resources could still be useful. thoughts?  Basically we trust users to tell us their intention via the labels, and if they haven't removed the sidecar injection labels, we don't assume it is ambient.

Comment (hanxiaop) [reply pkg/config/analysis/analyzers/util/in_mesh.go:56]: This also makes sense to me for the current situation. I'll revise it. Thanks!

**Comment by hanxiaop** [comment]
@GregHanson Based on @linsun 's review, I've excluded the situation when we have both injection and ambient labels. PTAL, thanks!

**Pull request merged**